src/subscript/interp_relperm/interp_relperm.py

```diff
@@ -101,146 +100,43 @@
-def _is_valid_eclipse_keyword_family(familychoice: int):
-    return familychoice in [1, 2]
-
-
-def get_cfg_schema() -> dict:
-    """
-    Defines the yml config schema
-    """
-    schema = {
-        MK.Type: types.NamedDict,
-        MK.ElementValidators: (_is_valid_table_entries,),
-        MK.Content: {
-            "base": {
-                MK.Type: types.List,
-                MK.Content: {
-                    MK.Item: {
-                        MK.Type: types.String,
-                        MK.ElementValidators: (_is_filename,),
-                    }
-                },
-            },
-            "low": {
-                MK.Type: types.List,
-                MK.Content: {
-                    MK.Item: {
-                        MK.Type: types.String,
-                        MK.ElementValidators: (_is_filename,),
-                    }
-                },
-            },
-            "high": {
-                MK.Type: types.List,
-                MK.Content: {
-                    MK.Item: {
-                        MK.Type: types.String,
-                        MK.ElementValidators: (_is_filename,),
-                    }
-                },
-            },
-            "pyscalfile": {
-                MK.Type: types.String,
-                MK.ElementValidators: (_is_filename,),
-                MK.AllowNone: True,
-            },
-            "result_file": {MK.Type: types.String},
-            "family": {
-                MK.Type: types.Number,
-                MK.Default: 1,
-                MK.ElementValidators: (_is_valid_eclipse_keyword_family,),
-            },
-            "delta_s": {MK.Type: types.Number, MK.Default: 0.01},
-            "interpolations": {
-                MK.Type: types.List,
-                MK.ElementValidators: (_is_valid_interpolator_list,),
-                MK.Content: {
-                    MK.Item: {
-                        MK.Type: types.NamedDict,
-                        MK.ElementValidators: (_is_valid_interpolator,),
-                        MK.Content: {
-                            "tables": {
-                                MK.Type: types.List,
-                                MK.Content: {MK.Item: {MK.Type: types.Integer}},
-                            },
-                            "param_w": {MK.Type: types.Number, MK.AllowNone: True},
-                            "param_g": {MK.Type: types.Number, MK.AllowNone: True},
-                        },
-                    }
-                },
-            },
-        },
-    }
-
-    return schema
-
-
-def parse_satfunc_files(filenames: List[str]) -> pd.DataFrame:
+class Interpolator(BaseModel):
+    tables: Optional[List[int]] = []
+    param_w: Optional[Annotated[float, Field(strict=True, ge=-1, le=1)]] = None
+    param_g: Optional[Annotated[float, Field(strict=True, ge=-1, le=1)]] = None
+
+    @model_validator(mode="after")
+    def check_param_w_or_param_g(self) -> "Interpolator":
```

Comment:
Adding `from __future__ import annotations` at the top will allow you just write `Interpolator` without quotes. This enables behaviour that will become default in some future version of Python (target used to be 3.12 but I'm not sure anymore.)

Reply:
fixed.